rvepak/mailroomupdated.py

```diff
@@ -0,0 +1,152 @@
+import os
+import sys
+import prettytable
+import datetime
+
+maxTriesCount = 0
+donors_history_file = 'donors_history.txt'
+
+
+def input_prompts():
+    '''
+    this method prompts for the 3 values
+    '''
+    global maxTriesCount
+    allowed_responses_list = sorted(allowed_response_map.keys())
+    response = input('Choose an action from the following {}: '.format(
+        allowed_responses_list))
+    if str(response) not in allowed_responses_list:
+        while maxTriesCount < 3:
+            print(
+                'Response {} is not allowed. Allowed responses are {}. Number of trials left = {}. Please try again.'.
+                format(
+                    str(response), allowed_responses_list,
+                    str(3 - maxTriesCount)))
+            maxTriesCount += 1
+            input_prompts()
+    else:
+        allowed_response_map[response]()
+
+
+def get_donors():
+    '''
+    this method lists the available donors
+    '''
+    donor_list = []
+    donor_dict = {}
+    with open(donors_history_file, 'r') as rF:
+        next(rF)
+        for ln in rF:
+            ln = ln.strip('\n')
+            lst = ln.split('|')
+            donor_name = lst[0].strip(' ')
+            donation_amt = lst[1]
+            donation_date = lst[2]
+            donor_list.append(lst[0])
```

Comment:
For the sake of readability think about using constants for indexing... something like this:
```
DONOR_NAME = 0
DONATION_AMOUNT = 1
DONATION_DATE = 2
```